packages/components/view-switcher/src/view-switcher.stories.tsx

```diff
@@ -0,0 +1,165 @@
+import type { ComponentProps } from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import ViewSwitcher from '.';
+import type { TViewSwitcherButtonProps } from './view-switcher-button';
+import { iconArgType } from '@/storybook-helpers';
+
+const BUTTONS_COUNT = 4;
+const DEFAULT_BUTTON_ICON = 'WorldIcon';
+
+type ViewSwitcherPropsAndCustomArgs = ComponentProps<
+  typeof ViewSwitcher.Group
+> & {
+  iconViewSwitcherButton1: string;
+  iconViewSwitcherButton2: string;
+  iconViewSwitcherButton3: string;
+  iconViewSwitcherButton4: string;
+  iconOnlyButton1: boolean;
+  iconOnlyButton2: boolean;
+  iconOnlyButton3: boolean;
+  iconOnlyButton4: boolean;
+  isDisabledButton1: boolean;
+  isDisabledButton2: boolean;
+  isDisabledButton3: boolean;
+  isDisabledButton4: boolean;
+};
+
+const meta = {
+  title: 'Components/ViewSwitcher',
+  component: ViewSwitcher.Group,
+  tags: ['autodocs'],
+} satisfies Meta<ViewSwitcherPropsAndCustomArgs>;
+
+export default meta;
+
+type Story = StoryObj<ViewSwitcherPropsAndCustomArgs>;
+
+export const Default: Story = {
+  args: {
+    isCondensed: false,
+    defaultSelected: 'Button 3',
+    iconViewSwitcherButton1: DEFAULT_BUTTON_ICON,
+    iconOnlyButton1: false,
+    isDisabledButton1: false,
+    iconViewSwitcherButton2: DEFAULT_BUTTON_ICON,
+    iconOnlyButton2: false,
+    isDisabledButton2: false,
+    iconViewSwitcherButton3: DEFAULT_BUTTON_ICON,
+    iconOnlyButton3: false,
+    isDisabledButton3: false,
+    iconViewSwitcherButton4: DEFAULT_BUTTON_ICON,
+    iconOnlyButton4: false,
+    isDisabledButton4: false,
+  },
+  argTypes: {
+    children: {
+      table: {
+        disable: true,
+      },
+    },
+    defaultSelected: {
+      table: {
+        disable: true,
+      },
+    },
+    ...[...Array(BUTTONS_COUNT).keys()].reduce((acc, index) => {
+      const i = index + 1;
+      const temp = {
+        ...acc,
+        ...{
+          [`iconViewSwitcherButton${i}`]: {
+            ...iconArgType,
+            table: {
+              subcategory: `View Switcher Button ${i}`,
+            },
+          },
+          [`iconOnlyButton${i}`]: {
+            table: {
+              subcategory: `View Switcher Button ${i}`,
+            },
+          },
+          [`isDisabledButton${i}`]: {
+            table: {
+              subcategory: `View Switcher Button ${i}`,
+            },
+          },
+        },
+      };
+      return temp;
+    }, {}),
```

Comment:
Sometimes I think a little repetition makes code easier to understand and thus, to maintain.

What do you think about something like this instead?
```js
   const categorize = (index: number) => ({ table: { subcategory: `View Switcher Button ${i}` } });
  
  ...

    iconViewSwitcherButton1: categorize(1),
    iconOnlyButton1: categorize(1),
    isDisabledButton1: categorize(1),
    iconViewSwitcherButton2: categorize(2),
    iconOnlyButton2: categorize(2),
    isDisabledButton2: categorize(2),
    iconViewSwitcherButton3: categorize(3),
    iconOnlyButton3: categorize(3),
    isDisabledButton3: categorize(3),
    iconViewSwitcherButton4: categorize(4),
    iconOnlyButton4: categorize(4),
    isDisabledButton4: categorize(4),
```

Reply:
Agreed. Done in 12591b3b82a56d88b5c5c9023d165b005ac0f6fa